### DISCUSSION TITLE
Help with install -  Ubuntu on Rasberry Pi - dependencies

### DISCUSSION THREAD

**Discussion by chiachain** [Chia-Network/chia-blockchain#1739] [Q&A]
Hello and thanks!

I am a beginner, so nothing is obvious to me.  I just installed a fresh Ubuntu on Rasberry Pi and did updates:  sudo apt-get update / sudo apt-get upgrade -y   and am now here:

# Checkout the source and install
git clone https://github.com/Chia-Network/chia-blockchain.git -b latest
cd chia-blockchain

I tried for instance:
 git clone https://github.com/Chia-Network/chia-blockchain.git -b latest   but option "b" is not understood.  Tried without the -b latest and it started but then "Unable to locate package clone" and "Unable to locate package https://github.com/Chia-Network"

I downloaded the GUI and tried to install that but get "Unable to install chia-blockchain: The following packages have unmet dependencies"

Appreciate your help!

Chris

**Comment by dorkmo** [comment]
you might try the experimental .deb file here: https://github.com/Chia-Network/chia-blockchain/wiki/Raspberry-Pi

**Reply by chiachain** [reply]
Yes, thanks, but that is what I am trying to do. But the directions are for those who know their way around these languages. I don't. So, can only follow the directions given but if some steps are skipped then I am stuck.  I have read that page very carefully about a dozen times and tried doing what it mentioned but it does not work.  I have about 12 hours into this, and I know that there is just one step that is missing. The GUI needs dependencies, whatever that is, and the commands give errors like I mentioned above.  I know that if someone who knows how to do it could do it in seconds, but I am just spinning my wheels. It is so hard.

**Comment by chiachain** [comment]
Does anyone know how to....       "Install dependencies for Ubuntu 20.04 LTS" 

and clarify this:......            # Checkout the source and install    git clone https://github.com/Chia-Network/chia-blockchain.git -b latest

I don't know what these commands need exactly.  So what I am asking for is EXACTLY what do I do with this.....

git clone https://github.com/Chia-Network/chia-blockchain.git -b latest
or
sudo git clone https://github.com/Chia-Network/chia-blockchain.git -b latest
or 
git clone https://github.com/Chia-Network/chia-blockchain.git 
or
do something else first and then run a command????

Thanks much!!!!  I give up trying to figure it out myself.

**Reply by depado** [reply]
It's fairly straightforward, open a terminal and follow the instructions by copy/pasting each line. 
As for Ubuntu 20.04 LTS, my guess is you have to install git and python3 before continuing the installation steps like so:

`sudo apt install python3 git`

The first line you pasted is the correct one, you shouldn't run this as a superuser (that's what `sudo` means - "super user do"):

`git clone https://github.com/Chia-Network/chia-blockchain.git -b latest`

**Reply by chiachain** [reply]
Thanks Depado,

That did the trick. Really appreciate your help.  I have it running now.  I tried to close the terminal and the GUI shut down.  I don't see any icon or way to start it so back to terminal again.  Had to run through the whole process again to get it back up.  Seems you can't close the terminal or it will shut down? Odd. Is there suppossed to be a shortcut anywhere or easy way to find it, or do you have to go to the terminal and through all the steps to get it back up?  I will just leave it running, but just curious. Thanks again.

**Reply by depado** [reply]
The only steps you need to do to start the UI are:

```sh
$ cd chia-blockchain
$ . ./activate
$ cd chia-blockchain-gui
$ npm run electron
```

You don't have to reinstall everything (no `sh install.sh` or `sh install-gui.sh`). You can have a look at how you can create bash functions or aliases to automate this process. I think there is also a GUI installer somewhere but can't remember how that works, sorry.

**Reply by chiachain** [reply]
Thanks so much for your help!!!

So.....now have the program up and running and connected.  Only trouble is is that it is not syching.  I did a restart, but no go.  

The only two connections showing are Farmer and Wallet (no Full Nodes).

Any advice on how to get it synching would be great! Thanks again!

**Comment by chiachain** [comment]
UI is up and running, and synched. Trouble is...not syncing at all. Only farmer and wallet connections showing. Any thoughts on how to fix this?  I think I will try the rasbian 64 os again.

**Reply by dorkmo** [reply]
Might be to port 8444 forwarding issue. Need to log in to you network router and point 8444 at your node

**Reply by chiachain** [reply]
Thanks. I was running two nodes with the same key, as one was plotting and farming while the other was just farming the finished plots.  I found that this was a problem. So I stopped the wifi connection on the plotter/farmer, and the farm only got synced up fast.  I tried to find the setting in my router, but not confident I did it right.  I then edited the config file to upnp=false on the plotter so it would not farm.

Phweew....learning a lot. Thanks for the help.
